platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/FilteredModuleComponentRepository.java

```diff
@@ -121,7 +121,7 @@ public void resolveArtifact(ComponentArtifactMetadata artifact, ModuleSources mo
         public MetadataFetchingCost estimateMetadataFetchingCost(ModuleComponentIdentifier moduleComponentIdentifier) {
             return whenModulePresent(moduleComponentIdentifier.getModuleIdentifier(), moduleComponentIdentifier,
                     () -> delegate.estimateMetadataFetchingCost(moduleComponentIdentifier),
-                    () -> MetadataFetchingCost.FAST);
+                    () -> MetadataFetchingCost.CHEAP);
```

Comment:
The naming system here really is terrible. `FAST` vs `CHEAP` vs `EXPENSIVE` means nothing to me at first glance. 

What this _really_ seems to mean is if something is FAST, we assume we definitely return the metadata immediately -- and therefore do not need to download it in parallel. EXPENSIVE means we definitely need to download the metadata in parallel. CHEAP means we don't know.

In the prior code, by returning `FAST` we essentially said "since this module is filtered from this repository, _we can definitely return the metadata immediately_". This is wrong. We don't know if we can return it immediately, so we should use CHEAP instead here so that we check the next repository.